src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -103,20 +120,91 @@ def _check_if_empty_ts(time_step: BindingConstraintFrequency, time_series: Optio
         time_series_length = (365 * 24 + 24) if time_step == BindingConstraintFrequency.HOURLY else 366
         return time_series if time_series is not None else pd.DataFrame(np.zeros([time_series_length, 1]))
 
-    def _write_binding_constraint_ini(self) -> None:
-        binding_constraints_ini_content = {
-            idx: idx_constraint.local_properties.list_ini_fields
-            for idx, idx_constraint in enumerate(self.binding_constraints.values())
-        }
-        self.ini_file.ini_dict = binding_constraints_ini_content
+    def _write_binding_constraint_ini(
+        self,
+        properties: BindingConstraintProperties,
+        constraint_name: str,
+        constraint_id: str,
+        terms: Optional[list[ConstraintTerm]] = None,
+    ) -> None:
+        """
+        Write or update a binding constraint in the INI file.
+
+        """
+
+        current_ini_content = self.ini_file.ini_dict_binding_constraints or {}
+
+        existing_section = next(
+            (section for section, values in current_ini_content.items() if values.get("name") == constraint_name),
+            None,
+        )
+
+        if existing_section:
+            existing_terms = current_ini_content[existing_section]
+
+            serialized_terms = (
+                {term.id: serialize_term_data(term.data, term.offset, term.weight) for term in terms} if terms else {}
+            )
+
+            existing_terms.update(serialized_terms)  # type: ignore
+            current_ini_content[existing_section] = existing_terms
+
+            # Persist the updated INI content
+            self.ini_file.ini_dict_binding_constraints = current_ini_content
+            self.ini_file.write_ini_file()
+        else:
+            terms_dict = (
+                {
+                    term.id: ConstraintTerm(data=term.data, offset=term.offset, weight=term.weight)
+                    if isinstance(term.data, (LinkData, ClusterData))
```

Comment:
Don't think you need this if else as these are the 2 only possibilities for term.data